css-conditional-5/Overview.bs

```diff
@@ -1067,6 +1087,159 @@ Style Container Features</h3>
 	are [=computed value|computed=] with respect to the [=query container=],
 	the same as other values.
 
+<h3 id="scroll-state-container">
+Scroll State Container Features</h3>
+
+	A <dfn export>container scroll-state query</dfn> allows to query a container for various
+	state that depends on scroll position. The ''scroll-state'' [=query container=] can either
+	be the scroller itself, or an element that is affected by the scroll position of an ancestor
+	[=scroll container's=] [=scrollport=].
+
+<h4 id="updating-scroll-state">
+Updating Scroll State</h4>
+	<div class=note>
+		Note: This section is subject to change as a result of resolving
+		<a href="https://github.com/w3c/csswg-drafts/issues/10796">issue #10796</a>
+	</div>
+
+	Scroll state may cause layout cycles since queried scroll state may cause style changes,
+	which may lead to scroll state changes as a result of layout. The same issue exists for
+	[=scroll progress timelines=], and scroll state is handled in a similar manner.
+
+	To avoid such layout cycles, ''scroll-state'' [=query containers=] update their
+	current state once as the last step of [=run the scroll steps=]. Then, after the
+	resizeObserver loop in the
+	<a href="https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-processing-model">HTML event loop processing model</a>,
+	the scroll state of every ''scroll-state'' [=query container=] is updated.
+	If that state has changed since the scroll state update in [=run the scroll steps=],
+	re-run the style and layout update if necessary.
```

Comment:
I find this a bit confusing.  Is the "current state" mentioned in the first sentence the same as the "scroll state" in the second sentence?

In the second sentence, avoid the passive voice by saying "update the scroll state" rather than "the scroll state ... is updated".

It's also not clear to me how this prevents cycles.  It sounds like this is saying to re-run an arbitrary number of times.  Or does it mean to re-run just once and then stop?

Reply:
> I find this a bit confusing. Is the "current state" mentioned in the first sentence the same as the "scroll state" in the second sentence?

Changed the latter "scroll state" to "current state" which should be clearer.

> In the second sentence, avoid the passive voice by saying "update the scroll state" rather than "the scroll state ... is updated".

Done.

> It's also not clear to me how this prevents cycles. It sounds like this is saying to re-run an arbitrary number of times. Or does it mean to re-run just once and then stop?

Just a single time. Clarified.